templates/traefik.yml.j2

```diff
@@ -85,4 +85,4 @@ providers:
     network: {{ devture_traefik_config_providers_docker_network | to_json }}
     endpoint: {{ devture_traefik_config_providers_docker_endpoint | to_json }}
   file:
-    filename: /config/provider.yml
+    directory: /config
```

Comment:
`/config` would contain:

- this new `custom.yml` file
- `provider.yml` generated based on `templates/provider.yml.j2`
- `traefik.yml` generated based on `templates/traefik.yml.j2`

Given that all 3 files are in the same directory and they're all `.yml` files, I hope the `file` provider will not try to load `traefik.yml` and try to choke on it. I suppose you've tested that it doesn't cause any troubles?